### PR TITLE
Add flag to override GRPC Authority Header

### LINKED ISSUE #5449: Add flag to override GRPC Authority Header
### Proposal

We would like to be able to pass a command-line flag to the KEDA Adapter to override the GRPC Authority Header in the requests it makes to the KEDA Operator. 

### Use-Case

The address we use for our envoy proxy is not in the SAN provided by TLS Certificate of the pod that comes from our CA. If we can override the Authority Header with host name we know will be in the SAN of the certificate returned from the Envoy Proxy, we won't need to disable TLS.

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aliaqel-stripe**
Add a flag to the adapter called `metrics-service-grpc-authority` to allow us to set the Authority Header on the GRPC requests so that it can be a host name that is present in the Certificate SAN issued to the keda-operator pod if the GRPC address differs from that.

Validated on my KEDA Deployment and verified this header allows us to use TLS with our service mesh addresses + our CA issued certificates.

### Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5449 

Relates to https://github.com/kedacore/keda-docs/pull/1302

### CODE REVIEW

**Comment by aliaqel-stripe** [comment]
@JorTurFer Looks like the FOSSA test failed after I updated to main, known failure?

**Comment by aliaqel-stripe** [comment]
> @JorTurFer Looks like the FOSSA test failed after I updated to main, known failure?

I'm going to guess a recent import that merged into main broke this test

**Review by JorTurFer — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ Here is an overview of all new **experimental** features:
 
 ### Improvements
 
+- **General**: Add command-line flag in Adapter to allow override of GRPC Authority Header ([#5449](https://github.com/kedacore/keda/issues/5449))
```
Comment (zroubalik) [inline CHANGELOG.md:65]:
```suggestion
- **General**: Add command-line flag in Adapter to allow override of gRPC Authority Header ([#5449](https://github.com/kedacore/keda/issues/5449))
```

nit

**Comment by aliaqel-stripe** [comment]
@zroubalik Adopted your comments, let me know how this looks and if we can merge soon so I can consume the CI built Image for this commit

**Review by zroubalik — APPROVED** [review]

**Pull request merged**